qiskit/opflow/gradients/derivative_base.py

```diff
@@ -86,19 +88,24 @@ def gradient_wrapper(self,
             operator: The operator for which we want to get the gradient, Hessian or QFI.
             bind_params: The operator parameters to which the parameter values are assigned.
             grad_params: The parameters with respect to which we are taking the gradient, Hessian
-                        or QFI. If grad_params = None, then grad_params = bind_params
+                or QFI. If grad_params = None, then grad_params = bind_params
             backend: The quantum backend or QuantumInstance to use to evaluate the gradient,
                 Hessian or QFI.
+            expectation: The expectation converter to be used. If none is set then
+                `PauliExpectation()` is used.
 
         Returns:
-            callable(param_values): Function to compute a gradient, Hessian or QFI. The function
+            Function to compute a gradient, Hessian or QFI. The function
             takes an iterable as argument which holds the parameter values.
         """
         from ..converters import CircuitSampler
         if not grad_params:
             grad_params = bind_params
 
         grad = self.convert(operator, grad_params)
+        if expectation is None:
+            expectation = PauliExpectation()
```

Comment:
We should select the expectation converter based on the backend, such that e.g. it uses Pauli expectation for shot-based but Matrix expectation for statevector simulation. So something like
```python
if expectation is None and backend is not None:  # if backend is None we don't need one
    expectation = ExpectationFactory.build(grad, backend)  # you can import this from qiskit.opflow too

if expectation is not None:
    grad = expectation.convert(grad)

Reply:
In that case, should we remove the `expectation` argument which would enable the user to make this choice?

Reply:
Actually, the expectation factory doesn't support finding the correct expectation converter if the operator is mixed (i.e. contains multiple primitives) as is the case for gradients. So I'm good with your solution.